src/Searching.js

```diff
@@ -0,0 +1,137 @@
+/*
+Copyright 2019 New Vector Ltd
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import EventIndexPeg from "./EventIndexPeg";
+import MatrixClientPeg from "./MatrixClientPeg";
+
+function serverSideSearch(term, roomId = undefined) {
+    let filter;
+    if (roomId !== undefined) {
+        filter = {
+        // XXX: it's unintuitive that the filter for searching doesn't have the same shape as the v2 filter API :(
+            rooms: [roomId],
+        };
+    }
+
+    const searchPromise = MatrixClientPeg.get().searchRoomEvents({
+        filter: filter,
+        term: term,
+    });
+
+    return searchPromise;
+}
+
+async function combinedSearchFunc(searchTerm) {
```

Comment:
Maybe drop the `Func`?

```suggestion
async function combinedSearch(searchTerm) {
```